website/source/guides/tls.html.md

```diff
@@ -0,0 +1,375 @@
+verify the identity of the server you're connecting to based on a domain name
+such as `nomadproject.io`, Nomad verifies the node you're connecting to is in
+the expected region and configured for the expected role (e.g.
+`client.us-west.nomad`).
+
+Configuring TLS can be unfortunately complex process, but if you used the
+[Getting Started Guide's Vagrantfile][Vagrantfile] or have [cfssl][] and Nomad
+installed this guide will provide you with a production ready TLS
+configuration.
+
+~> Note that while Nomad's TLS configuration will be production ready, key
+   management and rotation is a complex subject not covered by this guide.
+   [Vault][] is the suggested solution for key generation and management.
+
+## Creating Certificates
+
+The first step to configuring TLS for Nomad is generating certificates. In
+order to prevent unauthorized cluster access, Nomad requires all certificates
+be signed by the same Certificate Authority (CA). This should be a *private* CA
+and not a public one like [Let's Encrypt][letsencrypt] as any certificate
+signed by this CA will be allowed to communicate with the cluster.
+
+### Certificate Authority
+
+There are a variety of tools for managing your own CA, [like the PKI secret
+backend in Vault][vault-pki], but for the sake of simplicity in this guide
+we'll use [cfssl][]. You can generate a private CA certificate and key with
+[cfssl][]:
+
+```shell
+# Generate the CA's private key and certificate
+cfssl print-defaults csr | cfssl gencert -initca - | cfssljson -bare nomad-ca
+```
+
+The CA key (`nomad-ca-key.pem`) will be used to sign certificates for Nomad
+nodes and must be kept private. The CA certificate (`nomad-ca.pem`) contains
+the public key necessary to validate Nomad certificates and therefore must be
+distributed to every node that requires access.
+
+### Node Certificates
+
+Once you have a CA certifacte and key you can generate and sign the
+certificates Nomad will use directly. TLS certificates commonly use the
+fully-qualified domain name of the system being identified as the certificate's
+Common Name (CN). However, hosts (and therefore hostnames and IPs) are often
+ephemeral in Nomad clusters. They come and go as clusters are scaled up and
+down or outages occur. Not only would signing a new certificate per Nomad node
+be difficult, but using a hostname provides no security or functional benefits
+to Nomad. To fulfill the desired security properties (above) Nomad certificates
+are signed with their region and role such as:
+
+* `client.global.nomad` for a client node in the `global` region
+* `server.us-west.nomad` for a server node in the `us-west` region
+
+To create certificates for the client and server in the cluster from the
+[Getting Started guide][guide-cluster] with [cfssl][] create ([or
+download][cfssl.json]) the following configuration file as `cfssl.json` to
+increase the default certificate expiration time:
+
+```json
+{
```

Comment:
`:+1: would love this sort of thing in a style guide

(or in my dreams: `mdfmt`)